controllers/flowlogspipeline/flp_objects.go

```diff
@@ -358,24 +372,27 @@ func (b *builder) service(old *corev1.Service) *corev1.Service {
 	}
 	// In case we're updating an existing service, we need to build from the old one to keep immutable fields such as clusterIP
 	newService := old.DeepCopy()
+	newService.Spec.Selector = b.selector
+	newService.Spec.SessionAffinity = corev1.ServiceAffinityClientIP
```

Comment:
oh actually I guess this session affinity should be needed even without your work on kafka right?